esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6035,46 +6031,31 @@ void CWsDfuEx::getFileAccess(IEspContext &context, IUserDescriptor *udesc, SecAc
         }
         case CFileAccessRole_External:
         {
-            IArrayOf<IEspDFUPartLocations> dfuPartLocations;
-            IArrayOf<IEspDFUPartCopies> dfuPartCopies;
-            resp.setNumParts(getFilePartsInfo(context, df, req.getCluster(), dfuPartLocations, dfuPartCopies));
-            resp.setFilePartLocations(dfuPartLocations);
-            resp.setFileParts(dfuPartCopies);
+            IArrayOf<IEspDFUFilePart> dfuParts;
+            getFilePartsInfo(context, fileDesc, dfuParts);
+            accessInfo.setNumParts(df->numParts());
+            accessInfo.setFileParts(dfuParts);
             if (req.getReturnJsonTypeInfo() || req.getReturnJsonTypeInfo())
             {
                 MemoryBuffer binLayout;
                 StringBuffer jsonLayout;
                 if (!getRecordFormatFromRtlType(binLayout, jsonLayout, df->queryAttributes(), req.getReturnJsonTypeInfo(), req.getReturnJsonTypeInfo()))
                     getRecordFormatFromECL(binLayout, jsonLayout, df->queryAttributes(), req.getReturnJsonTypeInfo(), req.getReturnJsonTypeInfo());
                 if (req.getReturnJsonTypeInfo() && jsonLayout.length())
-                    resp.setRecordTypeInfoJson(jsonLayout.str());
+                    accessInfo.setRecordTypeInfoJson(jsonLayout.str());
                 if (req.getReturnBinTypeInfo() && binLayout.length())
-                    resp.setRecordTypeInfoBin(binLayout);
+                    accessInfo.setRecordTypeInfoBin(binLayout);
             }
             break;
         }
         default:
             throwUnexpected();
     }
 
-    // setup "expiryTime"
-    unsigned expirySecs = req.getExpirySeconds();
-    if (expirySecs > maxFileAccessExpirySeconds)
-        expirySecs = maxFileAccessExpirySeconds;
-    time_t now;
-    time(&now);
-    CDateTime expiryDt;
-    expiryDt.set(now + expirySecs);
-    StringBuffer expiryTimeStr;
-    expiryDt.getString(expiryTimeStr);
-
-    StringBuffer keyPairName;
-    getFileDafilesrvKeyName(keyPairName, *df);
-
-    StringBuffer metaInfo;
-    getFileMeta(metaInfo, *df, udesc, role, expiryTimeStr, keyPairName, req);
-    resp.setMetaInfoBlob(metaInfo);
-    resp.setExpiryTime(expiryTimeStr);
+    StringBuffer metaInfo, expiryTime;
+    getFileMeta(metaInfo, fileName, fileDesc, udesc, expiryTime, jobId, req);
+    accessInfo.setMetaInfoBlob(metaInfo.str());
+    accessInfo.setExpiryTime(expiryTime.str());
```

Comment:
no need for these .str()'s - operator overload will handle it.

Reply:
ok